src/Features/Core/Portable/QualifyMemberAccess/AbstractQualifyMemberAccessDiagnosticAnalyzer.cs

```diff
@@ -92,6 +92,12 @@ private void AnalyzeOperation(OperationAnalysisContext context, IOperation opera
                 return;
             }
 
+            // We shouldn't qualify if it is inside a property pattern
```

Comment:
why not?  say we have:  `x is { y: JToken`.  Wouldn't we want to qualify `JToken` if necessary?

Reply:
@CyrusNajmabadi in your case `JToken` should be a constant value, right? But we shouldn't qualify constants.

Reply:
>  But we shouldn't qualify constants.

Why not?  Say i want to write:  `y: X.Y.SomeConstant`.  So i write `y: Y`, hoping i can fully qualify `Y` to `X.Y`.

Reply:
@CyrusNajmabadi, This analizator looks if we can add `this` or `me` depending on language, am I right? I didn't find any option to qualify access to constant or static field/property.

Reply:
i see what you're saying.  i thought this was the FullyQualify fixer, not the Qualify-This/Me fixer.   That said, i'm still not a fan of this all-or-nothing approach.  I feel like the bug is that we expand on the left side of `x: y`.  There's no issue expanding on the RHS.

Reply:
@zaytsev-victor Would this suggestion be a relatively simple change for you to implement?

Reply:
@CyrusNajmabadi this won't affect right side.
for the following code
```csharp
    class P
    {
        int s = 10;

        void M1(Class c)
        {
            if (c is { i: [|s|] })
            {

            }
        }

        class Class
        {
            public int i { get; set; }
        }
    }
```
It still suggests to add ```this```. I'm not sure if we should fix or not since it is not valid code.

Reply:
i'm ok with this behavior.  can you ensure there is a test for both cases (a field/prop on either side of the property pattern)?